base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -521,7 +530,7 @@ where KM: TransactionKeyManagerInterface
             "Build transaction with Fee: {}. Change: {}. Output: {:?}", total_fee, change, change_output,
         );
         // Some checks on the fee
-        if total_fee < Fee::MINIMUM_TRANSACTION_FEE {
+        if total_fee < Fee::MINIMUM_TRANSACTION_FEE && !self.allow_zero_fees {
             return self.build_err("Fee is less than the minimum");
         }
```

Comment:
I would just have removed this, this is a stupid check anyway as we don't have a consensus rule that prevents 0 fees. And we min fee checks on the mempool that's individually set and checked,.